SYCL/ESIMD/regression/math_const_fix_test.cpp

```diff
@@ -0,0 +1,238 @@
+  }
+
+  if (std::fabs(std_result_cos - scalar_result_cos) > 0.00001f) {
+    std::cout << "Scalar test failed for cos for " << x << "." << std::endl;
+    return 1;
+  }
+
+  for (int i = 0; i < SIMD; ++i) {
+    if (scalar_result_sin != vector_result_sin[i]) {
+      std::cout << "Vector test failed for sin " << x << "." << std::endl;
+      return 1;
+    }
+
+    if (scalar_result_cos != vector_result_cos[i]) {
+      std::cout << "Vector test failed for cos for " << x << "." << std::endl;
+      return 1;
+    }
+  }
+
+  return 0;
+}
+
+int test_atan2(float x, float y) {
+  std::vector<float> vector_result_atan2_fast(SIMD);
+  std::vector<float> vector_result_atan2(SIMD);
+
+  float ha = x;
+  float hb = y;
+  float scalar_result_atan2_fast = 0;
+  float scalar_result_atan2 = 0;
+
+  {
+    sycl::queue queue;
+
+    sycl::buffer<float, 1> vector_buffer(vector_result_atan2_fast.data(),
+                                         vector_result_atan2_fast.size());
+    sycl::buffer<float, 1> scalar_buffer(&scalar_result_atan2_fast,
+                                         sycl::range<1>(1));
+    sycl::buffer<float, 1> vector_buffer1(vector_result_atan2.data(),
+                                          vector_result_atan2.size());
+    sycl::buffer<float, 1> scalar_buffer1(&scalar_result_atan2,
+                                          sycl::range<1>(1));
+
+    auto e = queue.submit([&](sycl::handler &cgh) {
+      sycl::accessor<float, 1, sycl_write> vector_out =
+          vector_buffer.get_access<sycl_write>(cgh);
+      sycl::accessor<float, 1, sycl_write> scalar_out =
+          scalar_buffer.get_access<sycl_write>(cgh);
+      sycl::accessor<float, 1, sycl_write> vector_out1 =
+          vector_buffer1.get_access<sycl_write>(cgh);
+      sycl::accessor<float, 1, sycl_write> scalar_out1 =
+          scalar_buffer1.get_access<sycl_write>(cgh);
+
+      auto kernel = ([=]() [[intel::sycl_explicit_simd]] {
+        using namespace sycl::ext::intel::esimd;
+
+        simd<float, SIMD> a = ha;
+        simd<float, SIMD> b = hb;
+
+        simd<float, SIMD> vector_result =
+            sycl::ext::intel::experimental::esimd::atan2_fast(a, b);
+        simd<float, 1> scalar_result =
+            sycl::ext::intel::experimental::esimd::atan2_fast(ha, hb);
+        simd<float, SIMD> vector_result1 =
+            sycl::ext::intel::experimental::esimd::atan2(a, b);
+        simd<float, 1> scalar_result1 =
+            sycl::ext::intel::experimental::esimd::atan2(ha, hb);
+
+        vector_result.copy_to(vector_out, 0);
+        scalar_result.copy_to(scalar_out, 0);
+        vector_result1.copy_to(vector_out1, 0);
+        scalar_result1.copy_to(scalar_out1, 0);
+      });
+
+      cgh.single_task<class TestAtan>(kernel);
+    });
+    queue.wait();
+  }
+
+  float std_result_atan = std::atan2(ha, hb);
+
+  if (std::fabs(std_result_atan - scalar_result_atan2) > 0.0001f) {
+    std::cout << "Scalar test failed for atan2 for " << x << "," << y << "."

```

Comment:
This and other error reports in this test create too many lines in the final LOG (hundreds or even thousands of them).
See here (grep for "failed for atan" : http://icl-jenkins.sc.intel.com:8080/blue/rest/organizations/jenkins/pipelines/LLVM-Test-Suite-CI-TMP/pipelines/LLVM-Test-Suite-CI-Linux/runs/3767/nodes/27/steps/177/log/?start=0

Can you please limit them with some fixed number (less than 100)?
As a variant each error can be still registered, and the first N of them reported and at the end reports the %pass.